File: lab09/linkedlist.c
//corey chen lab 107 11/16/2016
#include <stdio.h>
#include <stdlib.h>

struct node {
  int val;
  struct node *next;
};

int main() {
	int i = 0;
	struct node *head = NULL;
	head = malloc(sizeof(struct node));

	int numVal;
	printf("Enter how many values to input: ");
	scanf("%d", &numVal);

	struct node *tempNode = head;
	for(i = 0; i < numVal; i++) {
		int num;
		printf("Enter value %d: ", i+1);
		scanf("%d", &num);
		struct node *nextNode = malloc(sizeof (struct node));
		nextNode->val = num;
		tempNode->next = nextNode;
		tempNode = nextNode;
	}

	struct node *tempNode2 = head->next;
	for(i = 0; i < numVal; i++) {
		int num = tempNode2->val;
		printf("%d\n", num);
		tempNode2 = tempNode2->next;
	}

	
	for (i = 0; i < numVal; i++) {
		struct node *tempNode3 = head;
		//printf("%d\n", head->val);
		while(tempNode3->next->next != NULL) {
			tempNode3 = tempNode3->next;
		}
		printf("deleting node %d\n", tempNode3->next->val);
		free(tempNode3->next);
		tempNode3->next = NULL;
	}
	printf("deleting Dummy head\n");
	free(head);

return 0;
}	

File: lab12/linkedlist.m
#import <Foundation/NSObject.h>
#import <math.h>
#import <stdio.h>

@interface ListNode : NSObject
{
@private
   int val;
   ListNode *next;
}

- (id) val: (int) value;
- (int) val;
- (id) next: (ListNode*) nextNode;
- (ListNode*) next;
@end

@implementation ListNode

- (id) val: (int) value {
  val = value;
  return self;
}

- (int) val {
  return val;
}

- (id) next: (ListNode*) nextNode {
  next = nextNode;
  return self;
}

- (ListNode*) next {
  return next;
}

@end

int main(void) {
  int i = 0;
  ListNode *head = [ListNode new];
  
  int numVal;
  printf("Enter How many values to input: ");
  scanf("%d", &numVal);

  ListNode *tempNode = head;
  for(i = 0; i < numVal; i++) {
    int num;
    printf("Enter value %d: ", i+1);
    scanf("%d", &num);
    ListNode *nextNode = [ListNode new];
    [nextNode val:num];
    [tempNode next:nextNode];
    tempNode = nextNode;
  }

  ListNode *tempNode2 = [head next];
  for (i = 0; i < numVal; ++i) {
    int num = [tempNode2 val];
    printf("%d\n", num);
    tempNode2 = [tempNode2 next];
  }

  for (i = 0; i < numVal; i++) {
    ListNode *tempNode3 = head;
    while ([[tempNode3 next] next] != nil) {
      tempNode3 = [tempNode3 next];
    }
    printf("deleting node %d\n", [[tempNode3 next] val]);
    [[tempNode3 next] release];
    [tempNode3 next:nil];
  }
  printf("deleting Dummy head\n");
  [head release];


  return 0;

}







